src/platform/plugins/shared/visualizations/public/embeddable/visualize_embeddable.tsx

```diff
@@ -159,7 +158,7 @@ export const getVisualizeEmbeddableFactory: (deps: {
       linkedToLibraryArg: boolean
     ) => {
       return serializeState({
-        serializedVis: vis$.getValue().serialize(),
+        serializedVis: serializedVis$.value,
```

Comment:
This was the fix. `vis$.getValue().serialize()` serialized `data.aggs[index].params.timeRange` as dashboard timeRange instead of time range saved with vis.